opm/simulators/linalg/bda/openclSolverBackend.cpp

```diff
@@ -496,7 +513,7 @@ void openclSolverBackend<block_size>::initialize(int N_, int nnz_, int dim, doub
         d_Acols = cl::Buffer(*context, CL_MEM_READ_WRITE, sizeof(int) * nnzb);
         d_Arows = cl::Buffer(*context, CL_MEM_READ_WRITE, sizeof(int) * (Nb + 1));
 
-        wcontainer->init(wellContribs, N, Nb);
+        d_toOrder = cl::Buffer(*context, CL_MEM_READ_WRITE, sizeof(int) * Nb);
```

Comment:
Just wondering: Do we really need to write to `d_toOrder` in a kernel? Maybe we can use `CL_MEM_READ_ONLY`?

Reply:
I created this buffer with `CL_MEM_READ_WRITE` to follow what was done in the other buffers. But you are right, this buffer is not written to, that is, the values contained in it are not changed by the kernel. The same would apply to other buffers in the code, e.g., `d_Avals`, `d_Acols` and `d_Arows`. I don't know if there would be performance gains in changing them to `CL_MEM_READ_ONLY`, gotta look into that.

Reply:
we may revisit later